### ISSUE TITLE
Resource object's attributes are not used during resource creation

### ISSUE THREAD

**Issue by abourdon** [crnk-project/crnk-framework#511]
Hi,

Following the JSON:API specification, creating a resource must respect the [resource object schema](https://jsonapi.org/format/#document-resource-objects) that uses an `attributes` section to specify resource's attributes (https://jsonapi.org/format/#crud-creating). For instance:

```json
{
  "data": {
    "type": "project",
    "attributes": {
      "name": "What a beautiful project name"
    }
  }
}
```
However, during the resource creation, the `attributes` section seems to be ignored. Indeed, the project's name attribute returned after creation is `null`:

```json
{"data":{"id":"124","type":"project","name":null,"links":{"self":"http://localhost:8080/projects/124"}}}
```

Interesting case, it is possible to specify the project's name by adding it out of the `attributes` section, as the following:
```json
{
  "data": {
    "type": "project",
    "name": "What a beautiful project name"
  }
}
```

And now it works although it is not JSON:API compliant:
```json
{"data":{"id":"125","type":"project","name":"What a beautiful project name","links":{"self":"http://localhost:8080/projects/125"}}}
```

Thanks!

**Comment by remmeier** [comment]
two formats are supported, JDON:API and a slightly simplified flavor of it. i assume you have added the second ad dependeny? crnk-format-plain-json? you may remove that again or used it next to each other by specifying the content type while posting

**Comment by abourdon** [comment]
Hi,

Indeed I'm using the `crnk-format-plain-json` as defined in the [getting started](https://www.crnk.io/).
But when reading [documentation](https://www.crnk.io/releases/stable/documentation/#_formats) it says it is compatible also with the JSON:API if defining content type (so I presumed `application/vnd.api+json`) in the request. What I did but without success.

But it seems `crnk-format-plain-json` is also waiting for `Accept` header. Indeed when using both `Content-Type` AND `Accept` header it works. That's strange because only Content-Type should be necessary when requesting a JSON:API server: https://jsonapi.org/format/#content-negotiation.

Note that it is working well if disabling `crnk-format-plain-json`. In this case, only Content-Type is necessary as expected.

Regards

**Comment by remmeier** [comment]
yes would make sense to add that, thx for pointing it out! so far only the accept header is used in the initial selection i think. and content type then necessary as well
